### PR TITLE
Update demos (#11970)

### PR DESCRIPTION
**Pull request by namelessssssssssss**
## What is the purpose of the change

This pull request updates several demo use cases in the `dubbo-demo` module that were using deprecated APIs. 

## Brief changelog
- Remove org.apache.dubbo.demo.consumer.Application#runWithRefer()
- Remove org.apache.dubbo.demo.provider.Application#startWithExport()  
- Remove com.apache.dubbo.demo.graalvm.consumer.Application#runWithRefer() 
- Rename 'com' to 'org' 
- Remove org.apache.dubbo.demo.graalvm.provider.Application#startWithExport() 

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by MieAh — APPROVED** [review]

File: dubbo-demo/dubbo-demo-native/dubbo-demo-native-provider/pom.xml [hunk 1]
```diff
@@ -50,6 +51,10 @@
             <groupId>org.apache.dubbo</groupId>
             <artifactId>dubbo-registry-zookeeper</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-registry-nacos</artifactId>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (namelessssssssssss) [reply dubbo-demo/dubbo-demo-native/dubbo-demo-native-provider/pom.xml]: Thanks. I will fix it later.

**Review by AlbumenJ — COMMENTED** [review]
@chickenlj PTAL

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-demo/dubbo-demo-api/dubbo-demo-api-provider/src/main/java/org/apache/dubbo/demo/provider/Application.java [hunk 2]
```diff
@@ -47,23 +60,47 @@ private static void startWithBootstrap() {
...
+        serviceConfig.setRef(new DemoServiceImpl());
+
+        moduleConfigManager.addConfig(serviceConfig);
+
+        serviceConfig.export();
 
-        System.out.println("dubbo service started");
         new CountDownLatch(1).await();
     }
+
```
[the remark this answers is not in the document]
Comment (namelessssssssssss) [reply dubbo-demo/dubbo-demo-api/dubbo-demo-api-provider/src/main/java/org/apache/dubbo/demo/provider/Application.java:105]: Sure. I will fix it.

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**